interface.go

```diff
@@ -44,3 +45,12 @@ type SQSProducer interface {
 	BatchProduceMessage(ctx context.Context, messageBatchInput *sqs.SendMessageBatchInput) (*sqs.SendMessageBatchOutput, error)
 	ProduceMessage(ctx context.Context, messageInput *sqs.SendMessageInput) error
 }
+
+// SQSClient is an interface for interacting with the actual SQS resource in AWS
+type SQSClient interface {
```

Comment:
`aws-sdk-go-v2` removes all [`*iface`](https://docs.aws.amazon.com/sdk-for-go/v2/developer-guide/migrate-gosdk.html#mocking-and-iface) packages (including `sqsiface`) so we need to maintain one in this package.